core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2209,6 +2209,14 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
         if (!doubleSpendTxns.isEmpty()) {
             // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
             killTxns(doubleSpendTxns, tx);
+
+            // disconnect irrelevant inputs (otherwise might cause serialization issue)
+            for(TransactionInput input : tx.getInputs()) {
```

Comment:
Hmm, I'm not sure but I think there is a slight risk in doing this. Generally if two transactions are stored in the wallet and one spends the other it is expected that spentBy is set accordingly… but again I'm not sure. Would that be difficult to achieve with your patch?